### ISSUE TITLE
Infinite recursion when Copying EvaluationAttrValue

### ISSUE THREAD

**Issue by asenchenko** [taurus-org/taurus#848]
While adding python3 support to [taurus_pyqtgraph](https://github.com/taurus-org/taurus_pyqtgraph) i found that coping EvaluationAttrValue ( using copy module ) rises exception:
```
Traceback (most recent call last):
  File "taurustrendset.py", line 62, in <module>
    c2.setModel('eval:rand(2)')
  File "/home/user/taurus_pyqtgraph/taurus_pyqtgraph/taurustrendset.py", line 132, in setModel
    self._forceRead()
  File "/home/user/taurus_pyqtgraph/taurus_pyqtgraph/taurustrendset.py", line 340, in _forceRead
    value = copy.copy(value)
  File "/home/user/venv3/lib/python3.6/copy.py", line 106, in copy
    return _reconstruct(x, None, *rv)
  File "/home/user/venv3/lib/python3.6/copy.py", line 281, in _reconstruct
    if hasattr(y, '__setstate__'):
  File "/home/user/venv3/lib/python3.6/site-packages/taurus-4.4.1a0-py3.6.egg/taurus/core/evaluation/evalattribute.py", line 64, in __getattr__
    ret = getattr(self._attrRef, name)
  File "/home/user/venv3/lib/python3.6/site-packages/taurus-4.4.1a0-py3.6.egg/taurus/core/evaluation/evalattribute.py", line 64, in __getattr__
    ret = getattr(self._attrRef, name)
  File "/home/user/venv3/lib/python3.6/site-packages/taurus-4.4.1a0-py3.6.egg/taurus/core/evaluation/evalattribute.py", line 64, in __getattr__
    ret = getattr(self._attrRef, name)
  [Previous line repeated 327 more times]
RecursionError: maximum recursion depth exceeded while calling a Python object
```
Copy module uses `__copy__` method. If this method does not exist, it  falls back to pickle protocol (`__reduce__, __setstate__, __getstate__ etc.`).  `_reconstruct` function from copyreg module is used to create empty object (internally calls `__new__`).  After creating object `copy` checks, is there `__setstate__` method. `hasattr` invokes `__getattr__`. But there is no `_attrRef`, so it recursively invokes `__getattr__` again and again.  
For more details, see [copy.py](https://github.com/python/cpython/blob/3.6/Lib/copy.py) and [copyreg.py](https://github.com/python/cpython/blob/3.6/Lib/copyreg.py).

Simple code to reproduce error
```
import taurus
import copy
attr = taurus.Attribute('eval:rand(2)')
copy.copy(attr.read())
```
I believe, that we have same error with python2.

**Comment by cpascual** [comment]
I confirm the bug

> I believe, that we have same error with python2.

I just tested and I think it is fine with python2

**Comment by asenchenko** [comment]
> > I believe, that we have same error with python2. 
> I just tested and I think it is fine with python2

`hasattr` has difference in the behavior between python2 and python3.
In python3, `hasattr` catches `AttributeError` only and return `False` 
In python2,  it catches catches all exceptions and return `False`.
I think that there is a (small) performance penalty in python2.

**Comment by cpascual** [comment]
Note that this is essentially the same that we fixed for Tango in #831

**Comment by cpascual** [comment]
Fixed by #849
